src/components/Template/Template.tsx

```diff
@@ -1,23 +1,65 @@
 /** @jsx h */
 
 import { h, Component } from 'preact';
-import PropTypes from 'prop-types';
 import { renderTemplate, isEqual } from '../../lib/utils';
 
-class Template extends Component {
-  shouldComponentUpdate(nextProps) {
+type TemplateProps = {
+  data?: Record<string, any>;
+  rootProps?: Record<string, any>;
+  rootTagName?: string;
+  templateKey?: string;
+  templates?: Record<
+    string,
+    | string
+    | ((data: any, bindEvent: (...args: any[]) => string) => string)
+    | undefined
+  >;
+  templatesConfig?: {
+    helpers?: Record<
+      string,
+      (text: string, render: (value: any) => string) => string
+    >;
+    // https://github.com/twitter/hogan.js/#compilation-options
+    compileOptions?: {
+      asString?: boolean;
+      sectionTags?: Array<{
+        o?: string;
+        c?: string;
+      }>;
+      delimiters?: string;
+      disableLambda?: boolean;
+    };
+  };
+  useCustomCompileOptions?: Record<string, boolean | undefined>;
+  bindEvent?: (...args: any[]) => string;
+} & typeof defaultProps;
+
+const defaultProps = Object.freeze({
+  data: {},
+  rootTagName: 'div',
+  useCustomCompileOptions: {},
+  templates: {},
+  templatesConfig: {},
+});
+
+class Template extends Component<TemplateProps> {
```

Comment:
template should be generic with TData

Reply:
Tried it but I was not able to pass it without breaking everything :disappointed: 

I took another approach and used `PreparedTemplateProps` to create part of the props (but as a result `TTemplateData` is still set to `any`). We can work together on this to find a better solution if you want.

Reply:
we can iterate on this another time since the component is internal, possibly add a todo here?